pkg/cmd/pulumi/convert.go

```diff
@@ -337,19 +363,33 @@ func runConvert(
 		pCtx.Diag.Logf(sev, diag.RawMessage("", msg))
 	}
 
-	installProvider := func(provider tokens.Package) *semver.Version {
+	installPulumiProvider := func(provider tokens.Package) *semver.Version {
 		// If auto plugin installs are disabled just return nil, the mapper will still carry on
 		if env.DisableAutomaticPluginAcquisition.Value() {
 			return nil
 		}
 
+		if p, ok := packageBlockDescriptors[string(provider)]; ok && p.Name == "terraform-provider" {
+			if p.Parameterization == nil {
+				pCtx.Diag.Warningf(
+					diag.Message("", "terraform provider with no parameterization detected: %q"),
+					string(provider),
+				)
+				return nil
+			}
+			pCtx.Diag.Warningf(
+				diag.Message("", "no pulumi provider for %q, attempting to use locally bridged provider by generating an sdk"),
+				p.Parameterization.Name,
+			)
+		}
+
 		pluginSpec := workspace.PluginSpec{
 			Name: string(provider),
 			Kind: apitype.ResourcePlugin,
 		}
 		version, err := pkgWorkspace.InstallPlugin(pCtx.Base(), pluginSpec, log)
 		if err != nil {
-			pCtx.Diag.Warningf(diag.Message("", "failed to install provider %q: %v"), provider, err)
+			pCtx.Diag.Warningf(diag.Message("", "failed to install pulumi provider %q: %v"), provider, err)
```

Comment:
```suggestion
			pCtx.Diag.Warningf(diag.Message("", "failed to install provider %q: %v"), provider, err)
```
We only ever install pulumi providers, we don't need to distinguish here.